### PR TITLE
Fixes IPAM result in cni result

### PR DESCRIPTION
**Pull request by dougbtv**
Fixes IPAM result in cni result (which wasn't showing the result.ips object). Also removes duplicated interfaces assignment, just assigns it during initialization of the result object.

Previously, the IP addresses weren't present. This is a requirement for an IPAM CNI plugin, and in turn it was causing issues for other components that rely on having this IPAM result present in a Multus network-status results annotation (especially, Whereabouts IP Reconciliation which was clearing out assigned IP addresses due to this not being present)

### CODE REVIEW

File: bond/bond.go [hunk 1]
```diff
@@ -351,30 +351,35 @@ func cmdAdd(args *skel.CmdArgs) error {
...
-		for _, ipc := range result.IPs {
+		for _, ipc := range ipamResult.IPs {
 			// All addresses belong to the vlan interface
 			ipc.Interface = current.Int(0)
 		}
 
-		result.Interfaces = []*current.Interface{bondInterface}
+		result.IPs = ipamResult.IPs
+		result.Routes = ipamResult.Routes
+		result.DNS = ipamResult.DNS
```
Comment (maiqueb) [inline bond/bond.go:382]: this one is being overwritten below, in line 391 by whatever sits at `bondConf.DNS` . 

Which one of those should take precedence ?...

I recommend leaving this attribute out of the fray, focus on the issue at hand (missing IPs / routes), and discuss what should be done regarding DNS in a follow up PR / issue.

**Review by maiqueb — CHANGES_REQUESTED** [review]
I agree the resulting routes and IPs from the IPAM plugin should be carried back to the result. 

What I a bit unsure of is what should be done with the DNS configuration: IIUC, the IPAM would apply the `bondConf.DNS` , which would be reflected in the CNI result - hence, your code should also delete line 391.

**Review by maiqueb — APPROVED** [review]
Thank you.

**Comment by mmirecki** [comment]
lgtm

**Comment by mmirecki** [comment]
@killianmuldoon  @Eoghan1232 could you please take a look?

**Review by Eoghan1232 — APPROVED** [review]

**Pull request merged**